File: functions/lib/functions/src/main/java/com/functions/events/utils/EventIdGenerator.java
package com.functions.events.utils;

import com.github.f4b6a3.ulid.UlidCreator;

public final class EventIdGenerator {
    /** Must match the frontend `newEventId` helper: `evt_` plus a 26-character ULID. */
    private static final String EVENT_ID_PREFIX = "evt_";

    private EventIdGenerator() {
    }

    public static String newEventId() {
        return EVENT_ID_PREFIX + UlidCreator.getMonotonicUlid();
    }
}

File: frontend/services/src/events/eventsUtils/eventIdGenerator.ts
import { EventId } from "@/interfaces/EventTypes";
import { monotonicFactory } from "ulid";

/** Must match Java `EventIdGenerator`: `evt_` plus a 26-character ULID. */
const EVENT_ID_PREFIX = "evt_";
const nextUlid = monotonicFactory();

export function newEventId(): EventId {
  return `${EVENT_ID_PREFIX}${nextUlid()}` as EventId;
}
